service/config_provider.go

```diff
@@ -144,21 +144,22 @@ func (cm *configProvider) Get(ctx context.Context, factories component.Factories
 		return nil, fmt.Errorf("cannot close previous watch: %w", err)
 	}
 
-	ret, err := cm.mergeRetrieve(ctx)
+	retMap, closer, err := cm.mergeRetrieve(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("cannot retrieve the configuration: %w", err)
 	}
-	cm.closer = ret.CloseFunc
+
+	cm.closer = closer
```

Comment:
Why not `ret.Close` instead? (so that you don't have to change the return type of `mergeRetrieve`)